pilz_robot_programming/test/integrationtests/tst_sequence_with_gripper.py

```diff
@@ -0,0 +1,130 @@
+
+    def tearDown(self):
+        if hasattr(self, 'robot'):
+            self.robot._release()
+            self.robot = None
+
+    def test_only_gripper_in_sequence(self):
+        """ Tests a sequence with only a gripper command.
+        """
+        start_joint_values = [0, 0.5, 0.5, 0, 0, 0]
+        self.robot.move(Ptp(goal=start_joint_values))
+
+        goal_joint = 0.02
+        seq = Sequence()
+        seq.append(Gripper(goal=goal_joint))
+        self.robot.move(seq)
+
+        # Check that gripper has moved
+        current_joints = self.robot.get_current_joint_states(self._GRIPPER_GROUP_NAME)
+        self.assertEqual(1, len(current_joints))
+        self.assertAlmostEqual(goal_joint, current_joints[0])
+
+        # Check that robot has NOT moved
+        self.assertTrue(numpy.allclose(self.robot.get_current_joint_states(),
+                                       start_joint_values, atol=self._TOLERANCE_POSITION_COMPARE))
+
+    def test_invalid_gripper_execution(self):
+        """ Tests a sequence containing an invalid gripper command.
+        """
+        start_joint_values = [0, 0.5, 0.5, 0, 0, 0]
+        self.robot.move(Ptp(goal=start_joint_values))
+
+        gripper_open = 0.02
+        gripper_invalid_pos = 1
+
+        seq = Sequence()
+        seq.append(Gripper(goal=gripper_open))
+
+        seq.append(Lin(goal=Pose(position=Point(0.2, 0, 0.7))), blend_radius=0.01)
+        seq.append(Lin(goal=Pose(position=Point(0.2, 0.1, 0.7))))
+
+        seq.append(Gripper(goal=self._GRIPPER_POSE_CLOSED))
+        seq.append(Gripper(goal=gripper_invalid_pos))
+
+        self.assertRaises(RobotMoveFailed, self.robot.move, seq)
+
+        # Check that robot has NOT moved
+        self.assertTrue(numpy.allclose(self.robot.get_current_joint_states(),
+                                       start_joint_values, atol=self._TOLERANCE_POSITION_COMPARE))
+
+    def test_several_gripper_cmds_in_sequence(self):
+        """ Tests a sequence command which contains several gripper commands.
+        """
+        start_joint_values = [0, 0.5, 0.5, 0, 0, 0]
+        self.robot.move(Ptp(goal=start_joint_values))
+
+        gripper_open = 0.02
+        gripper_totally_open = 0.029
+
+        seq = Sequence()
+        seq.append(Gripper(goal=gripper_open))
+
+        seq.append(Lin(goal=Pose(position=Point(0.2, 0, 0.7))), blend_radius=0.01)
+        seq.append(Lin(goal=Pose(position=Point(0.2, 0.1, 0.7))))
+
+        seq.append(Gripper(goal=self._GRIPPER_POSE_CLOSED))
+        seq.append(Gripper(goal=gripper_totally_open))
+
+        self.robot.move(seq)
+
+        # Check that gripper has opened
+        curr_gripper_joints = self.robot.get_current_joint_states(self._GRIPPER_GROUP_NAME)
+        self.assertEqual(1, len(curr_gripper_joints))
+        self.assertAlmostEqual(gripper_totally_open, curr_gripper_joints[0])
+        # Check that robot has moved
+        self.assertFalse(numpy.allclose(self.robot.get_current_joint_states(),
```

Comment:
You could also compare the current pose to the expected pose `position=Point(0.2, 0.1, 0.7)`.

Reply:
Done.